### DISCUSSION TITLE
Issue running WASM app in 11.1 beta 2

### DISCUSSION THREAD

**Discussion by dbriard** [AvaloniaUI/Avalonia#15730] [Q&A] [answered]
Hi, I am trying to upgrade my app to .NET8 and 11.1 beta 2 and I am having issues.

To debug, I tried to create a new app, but even with a new app, I cannot run the browser version.

What I've tried:

1) I updated workloads, and installed latest avalonia template
2) I created a new project targeting beta1 and .NET8 using: 
dotnet new avalonia.xplat -av 11.1.0-beta1 -f net8.0
3) With beta1, I can run the browser project and see the textblock.
4) Upgrade all nuget avalonia packages to beta2
5) With beta2, I cannot run the sample anymore. I get the following error:
![image](https://github.com/AvaloniaUI/Avalonia/assets/3595443/a2992c7c-1836-40b4-bfa9-49747aff7b62)

Is WASM app supposed to work with beta2 and .NET 8? 

I also have another issue with my current app.

I have that line at the begining of Main()
await JSHost.ImportAsync("myapp/helpers.js", "./helpers.js");

helpers.js was in AppBundle in 11.0.10, now I moved it in wwwroot
but the browser's inspector show that it cannot locate the file.

Could someone tell me the good way to add interop js files in a avalonia project? Where to put the file so it can be found by JSHost.ImportAsync ?

Thanks for helping!

**Comment by maxkatz6** [comment]
Looks like you have either mixed versions of Avalonia dependencies, or (most likely) some of them were cached by the browser.

**Reply by dbriard** [reply]
Thanks @maxkatz6 I deleted the bin and obj folder and it worked for the new sample app.

However, not yet for my old App. I got a white screen after the splashscreen fade out. Probably an exception somewhere, however I cannot debug the app. There is no error in the Inspector.

I recreated a new project and copied my old code to it, as I was not able to compile my old Web project after migration from 11.0.10/net7 to 11.1beta2/net8... On my old project, Avalonia.Browser namespace cannot be found, I don't know why???? The nuget package is present. Do you have an idea about that?

Also another issue, with my old project (net7/11.0.10), when I run the app in Debug, it open the browser with the localhost url and the breakpoints worked well (with Edge only). Now, with the new project, running in Debug only open the browser without the URL, I had to copy it by hand, but the breakpoints are not hit. Is there something I can do to make Debugging work?

FI: I found the issue with JSHost.ImportAsync("...", "./helpers.js"); I had to replace ./ with ../ as now there is a _framework folder.

**Reply by jp2masa** [reply]
The debugging issue is probably #14570.

**Reply by dbriard** [reply]
Sorry my late reply, but you right, it look like the same issue with Debug ! :S

**Comment by dbriard** [comment]
Hi again @maxkatz6, @jp2masa,
Do you know if there is a way (workaround) to debug WASM app with the debugging issue here: https://github.com/AvaloniaUI/Avalonia/discussions/14570
Thanks for helping

**Reply by maxkatz6** [reply, marked as answer]
Try adding 
```
    <DebuggerSupport>true</DebuggerSupport>
    <WasmDebugLevel>5</WasmDebugLevel>
```

to your csproj

**Reply by dbriard** [reply]
> Try adding
> 
> ```
>     <DebuggerSupport>true</DebuggerSupport>
>     <WasmDebugLevel>5</WasmDebugLevel>
> ```
> 
> to your csproj

Still cannot get breakpoints working. Do you know if there is a way to write to the browser's inspector console?

**Reply by dbriard** [reply]
I found Trace.Listeners.Add(new ConsoleTraceListener()); in your samples Catalog project, that should help!
